ethereumj-core/src/main/java/org/ethereum/core/BlockchainImpl.java

```diff
@@ -726,7 +726,6 @@ private boolean isValid(Repository repo, Block block) {
         boolean isValid = true;
 
         if (!block.isGenesis()) {
-            isValid = isValid(block.getHeader());
```

Comment:
May be it makes sense to leave this check if it's not too heavy?
Or leave it here by default and add a flag to skip the check as soon as it is not needed?